.github/workflows/ci.yaml

```diff
@@ -63,6 +63,61 @@ jobs:
           bash ./ci/run_tests.sh
           bash ./ci/ensure_copyright.sh
 
+  ubuntu-clang14-address-undefined-leak:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          ssh-key: ${{ secrets.SSH_KEY }}
+
+      - name: Install clang-14
+        run: |
+          sudo apt-get update && \
+          sudo apt-get install build-essential software-properties-common -y && \
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y && \
+          sudo apt-get update && \
+          sudo apt-get install clang-14 && \
+          gcc -v
+      - name: Run build
+        env:
+          CC: clang-14
+          CXX: clang++-14
+          TESTENV: clang14
+        run: |
+          bash ./ci/run_address_sanitizer_tests.sh
+
+  # Memory sanitizer is not run because it requires building a custom
+  # instrumented libc.
+
+  ubuntu-clang14-thread:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          ssh-key: ${{ secrets.SSH_KEY }}
+
+      - name: Install clang-14
```

Comment:
We might be able to replace this with a premade action, such as:
https://github.com/marketplace/actions/install-llvm-and-clang

Maybe in a separate PR.

Reply:
Feel free to make a ticket for the future -- for now I think I'll just keep it simple.